### ISSUE TITLE
MLKit selfie segmentation increases the size of android app

### ISSUE THREAD

**Issue by A-KVerma** [googlesamples/mlkit#387]
#386

**Comment by A-KVerma** [comment]
Yes 50MB means adding all the 4 ABIs together, the size is between (15-20)MB for each ABIs.
Got it thanks for your reply, are you guys planning to further reduce the shared library libxeno_native size in near future ?

**Comment by steven-chengji-yan** [comment]
The size of libxeno_native is something we are aware of and would like to improve. However, we do not have a concrete plan to fix it very soon.

Currently, for one ABI, the compressed the size of the aar is 4+MB. Is the current size a blocker for you to use it? What size are you looking for?

**Comment by A-KVerma** [comment]
Even if we are creating an app bundle with MLKit selfie segmentation, one architecture type library will get downloaded which is approx 17MB. It is not a blocker as such but still it will be helpful if we can reduce it further as small as possible.

Goal : APK size must have minimum increase if we are adding this feature to our App

**Comment by steven-chengji-yan** [comment]
I just did some test with the latest selfie SDK and collected the following numbers:

- Use Android Studio to create an app with empty Activity. Do debug build (one ABI). Use [APK analyzer](https://developer.android.com/studio/debug/apk-analyzer) to check apk size: Raw File Size is 3.3MB, while Download Size is 2.7MB.
- Add ML Kit Selfie SDK dependency to the app. Do debug build (one ABI). Analyze the apk: Raw File Size is 18.8MB, while Download Size is 7.7MB.

The size I talked before is the compressed/download size. Here the diff is 7.7-2.7=5MB (including the native lib and other assets).
I assume the size you talked about is the uncompressed size. Here the diff is 18.8-3.3=15.5MB, which roughly matches the number you provided.

But anyway, I got your point about the size concerns. Thanks for the feedback.
